pkg/networkmanager/network-interface.jsx

```diff
@@ -123,7 +123,7 @@ export const NetworkInterfacePage = ({
     }
 
     function connect() {
-        if (iface.MainConnection && !(dev && ghostSettings))
+        if (! (iface.MainConnection || (dev && ghostSettings)))
```

Comment:
[before the react rewrite](https://github.com/cockpit-project/cockpit/blob/842d312dd90261046a1fef69477ee92dac254568/pkg/networkmanager/interfaces.js#L2332) the condition was:

    if (!self.main_connection && !(self.dev && self.ghost_settings)) {

So this effectively goes back to that.

Reply:
Was it? Expand pkg/networkmanager/interfaces.js  to see it: https://github.com/cockpit-project/cockpit/commit/5c7b284a082edc4c9bd66a1efd662bcfdd2b1dc7#diff-e6968ba34dad066826484f30b54426b4891fbd5e72d45297d295a8f46086e3c8L2332

Reply:
Ah right, I just saw the missing leading exclamationmark.